### PR TITLE
feat: Kick command

### LINKED ISSUE #21: Allow mentors to remove users from queue
`$kick <topic> <mention>`: Remove a user from a queue

Useful for moderation

### PR DESCRIPTION
**Pull request by skyegallup**
Closes #21 

Soft-blocked by #9: we'll either have to let that get merged first and then add bot responses for this command, or merge this first and add bot responses in that branch

### CODE REVIEW

File: src/main/java/listeners/MainEventListener.java [hunk 1]
```diff
@@ -253,7 +257,43 @@ private void showQueue(Member member, TextChannel channel, Server server, String
...
 
-    private void clear(Member member, TextChannel channel, Server server, String[] args) {
+    private void kick(Member member, TextChannel channel, Server server, String[] args, Member[] mentions) {
+        Member mentee = mentions[0];  // also takes up args[0]
+        String topicName = args[1];
+        String reason = Stream.of(args).skip(2).collect(Collectors.joining(" "));
+
+        // do not run if topic does not exist
+        Optional<Topic> optionalTopic = checkTopicExists(member, channel, server, topicName);
+        if (optionalTopic.isEmpty()) return;
```
Comment (peterkos) [inline src/main/java/listeners/MainEventListener.java:285]: I really like this flow!

File: src/main/java/listeners/MainEventListener.java [hunk 2]
```diff
@@ -253,7 +257,43 @@ private void showQueue(Member member, TextChannel channel, Server server, String
...
+        Topic topic = optionalTopic.get();
+        if (!isMentor(member, topic) && !isAdmin(member)) {
+            channel.sendMessage(member.getAsMention() + " You do not have permission to run this command.").queue();
+            return;
+        }
+
+        // do not run if mentee is not in the specified queue
+        if (!topic.isInQueue(mentee)) {
+            channel.sendMessage(String.format(
+                "%s User \"%s\" is not in the queue for topic \"%s\".",
```
Comment (peterkos) [inline src/main/java/listeners/MainEventListener.java:297]: In future might be good to factor these to -- oh wait, @sjmiller7 are these the messages you're currently factoring out to a static class?

File: src/main/java/listeners/MainEventListener.java [hunk 3]
```diff
@@ -97,6 +98,7 @@ public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
 
         Member member = Objects.requireNonNull(event.getMember());
         TextChannel channel = event.getChannel();
+        Member[] mentions = event.getMessage().getMentionedMembers().toArray(new Member[0]);
```
Comment (peterkos) [inline src/main/java/listeners/MainEventListener.java:114]: What is the purpose of this? Is this just so we can access the mentee in `kick`?

**Review by peterkos — APPROVED** [review]
LGTM :shipit:

File: src/main/java/listeners/MainEventListener.java (hunk 2 above)
Comment (skyegallup) [reply src/main/java/listeners/MainEventListener.java:297]: Yep, they are

File: src/main/java/listeners/MainEventListener.java (hunk 3 above)
Comment (skyegallup) [reply src/main/java/listeners/MainEventListener.java:114]: Yep. Right now, if I have a command `$kick @cheesegreater#0001`, `args[0] == "@cheesegreater`. It's really hard to convert that string to a Member, especially if people have identical usernames, so this allows us to access the Member objects in `kick()`.

**Review by Madrugaur — APPROVED** [review]

**Pull request merged**